admin/app/AdminAppLoader.scala

```diff
@@ -0,0 +1,68 @@
+import com.softwaremill.macwire._
+import common._
+import common.Logback.LogstashLifecycle
+import common.dfp.DfpAgentLifecycle
+import conf.switches.SwitchboardLifecycle
+import conf.{CachedHealthCheckLifeCycle, CommonGzipFilter}
+import controllers.HealthCheck
+import _root_.dfp.DfpDataCacheLifecycle
+import _root_.http.AdminHttpErrorHandler
+import model.{ApplicationIdentity, AdminLifecycle}
+import ophan.SurgingContentAgentLifecycle
+import play.api.http.HttpErrorHandler
+import play.api.inject.{NewInstanceInjector, SimpleInjector, Injector}
+import play.api.libs.ws.ning.NingWSComponents
+import play.api.mvc.EssentialFilter
+import play.api.routing.Router
+import play.api._
+import services.ConfigAgentLifecycle
+import router.Routes
+
+import scala.concurrent.ExecutionContext
+
+class AdminAppLoader extends ApplicationLoader {
+  override def load(context: ApplicationLoader.Context): Application = {
+    Logger.configure(context.environment)
+    val components = new BuiltInComponentsFromContext(context) with RoutingComponents
+    components.startLifecycleComponents()
+    components.application
+  }
+}
+
+trait AppComponents extends BuiltInComponents with NingWSComponents {
+  lazy val appIdentity = ApplicationIdentity("frontend-admin")
+  lazy val appMetrics = ApplicationMetrics()
+  implicit lazy val executionContext: ExecutionContext = actorSystem.dispatcher
+  // this is a workaround to make wsapi available to the injector
+  override lazy val injector: Injector = new SimpleInjector(NewInstanceInjector) + router + crypto + httpConfiguration + wsApi
```

Comment:
Because the code still uses `Ws.url` which is then making use of the injector to resolve `wsClient`. So as long as _some_ components aren't passed upon creation, we need that thing here.

To be honest I think it's almost a bug in play 2.4, but I don't see them fixing it now they're working on play 2.6, even if I report it.